native_libs/src/Python/PythonInterpreter.cpp

```diff
@@ -6,11 +6,82 @@
 #include "IO/IO.h"
 #include <cstdlib>
 
+#ifndef _WIN32
+#include <boost/filesystem.hpp>
+#endif
+
 #ifdef __linux__
 #include <dlfcn.h> // for dlopen
-#include <boost/filesystem.hpp>
 #endif
 
+#ifdef __linux__
+boost::filesystem::path loadedLibraryPath(std::string_view libraryName)
+{
+    auto pid = getpid();
+    auto mapsIn = openFileToRead(fmt::format("/proc/{}/maps", pid));
```

Comment:
```suggestion
    auto mapsIn = openFileToRead("/proc/self/maps");
```
/proc/self always refer to current pid, so there's no need to get it beforehand